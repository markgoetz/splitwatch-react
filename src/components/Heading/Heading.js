import React from 'react';
import PropTypes from 'prop-types';
import styles from './Heading.module.css';

export default function Heading(props) {
  const tagProps = { ...props };
  delete tagProps.children;
  delete tagProps.level;

    switch (props.level) {
        case '1':
            return <h1 {...tagProps} className={styles[`level${props.level}`]}>{ props.children }</h1>;
        case '2':
            return <h2 {...tagProps} className={styles[`level${props.level}`]}>{ props.children }</h2>;
        case '3':
            return <h3 {...tagProps} className={styles[`level${props.level}`]}>{ props.children }</h3>;
        case '4':
            return <h4 {...tagProps} className={styles[`level${props.level}`]}>{ props.children }</h4>;
        case '5':
            return <h5 {...tagProps} className={styles[`level${props.level}`]}>{ props.children }</h5>;
        case '6':
            return <h6 {...tagProps} className={styles[`level${props.level}`]}>{ props.children }</h6>;
        default:
            return;
    }
}

Heading.propTypes = {
    level: PropTypes.oneOf(['1', '2', '3', '4', '5', '6'])
};