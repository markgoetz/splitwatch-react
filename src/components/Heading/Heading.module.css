.level1, .level2, .level3, .level4, .level5, .level6 {
  font-family: var(--font-sans);
  font-weight: bold;
}

.level1 {
  font-size: 28px;
}

.level2 {
  font-size: 24px;
}

.level3 {
  font-size: 22px;
}

.level4 {
  font-size: 20px;
}

.level5 {
  font-size: 18px;
}

.level6 {
  font-size: 16px;
}