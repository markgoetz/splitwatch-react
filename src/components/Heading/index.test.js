import React from 'react';
import Heading from './Heading';
import { shallow } from 'preact-render-spy';

describe('Heading Component', () => {
  test('Level 1 renders correctly', () => {
    const context = shallow(<Heading level="1">Level 1</Heading>);
    expect(context).toMatchSnapshot();
  });

  test('Level 2 renders correctly', () => {
    const context = shallow(<Heading level="2">Level 2</Heading>);
    expect(context).toMatchSnapshot();
  });

  test('Level 3 renders correctly', () => {
    const context = shallow(<Heading level="3">Level 3</Heading>);
    expect(context).toMatchSnapshot();
  });

  test('Level 4 renders correctly', () => {
    const context = shallow(<Heading level="4">Level 4</Heading>);
    expect(context).toMatchSnapshot();
  });

  test('Level 5 renders correctly', () => {
    const context = shallow(<Heading level="5">Level 5</Heading>);
    expect(context).toMatchSnapshot();
  });

  test('Level 6 renders correctly', () => {
    const context = shallow(<Heading level="6">Level 6</Heading>);
    expect(context).toMatchSnapshot();
  });


});