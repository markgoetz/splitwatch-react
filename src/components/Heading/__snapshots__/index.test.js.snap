// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component Level 1 renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<h1>Level 1</h1>

`;

exports[`Heading Component Level 2 renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<h2>Level 2</h2>

`;

exports[`Heading Component Level 3 renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<h3>Level 3</h3>

`;

exports[`Heading Component Level 4 renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<h4>Level 4</h4>

`;

exports[`Heading Component Level 5 renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<h5>Level 5</h5>

`;

exports[`Heading Component Level 6 renders correctly 1`] = `
preact-render-spy (1 nodes)
-------
<h6>Level 6</h6>

`;
